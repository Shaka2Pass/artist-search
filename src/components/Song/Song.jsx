import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Song = ({ artist, trackTitle, lyrics }) => (
  <section>
    <h2><Link to={`/lyrics/${artist}/${trackTitle}`}></Link></h2>
    <h2>{trackTitle}</h2>
    <h3>{lyrics}</h3>
  </section>
);

Song.propTypes = {
  artist: PropTypes.string.isRequired,
  trackTitle: PropTypes.string.isRequired,
  lyrics: PropTypes.string.isRequired
};

export default Song;
