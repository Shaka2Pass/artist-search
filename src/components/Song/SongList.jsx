import Song from '../Song/Song.jsx';
import withList from '../withList';

const SongsList = withList(Song, 'songResults');

export default SongsList;
