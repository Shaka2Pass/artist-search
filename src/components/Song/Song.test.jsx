import React from  'react';
import { shallow } from 'enzyme';
import Song from './Song.jsx';

describe('Song', () => {
  it('Matches a snapshot of the Song component', () => {
    const wrapper = shallow (<Song
      lyrics="LALALALALALALA"
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
