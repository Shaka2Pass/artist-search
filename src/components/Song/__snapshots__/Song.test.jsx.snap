// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song Matches a snapshot of the Song component 1`] = `
<section>
  <h2>
    <Link
      to="/lyrics/U2/Gloria"
    />
  </h2>
  <h2>
    Gloria
  </h2>
  <h3>
    LALALALALALALA
  </h3>
</section>
`;
